### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by backus**
This PR doesn't fix all warnings, just a warning in expression.rb (which I mentioned in #18) and indentation warnings (mentioned in #10).

### CODE REVIEW

**Comment by backus** [comment]
@ammar could you review this PR?

**Comment by mbj** [comment]
@ammar To give this a bit of context. Mutant runs itself warning free, including its dependencies, we'd like to keep this property.

**Comment by ammar** [comment]
@backus Please excuse the delayed response. I just returned from a week of camping (Wadi Rum).

Thanks for this. Definitely something that needs fixing. I have a bit of catch up to do but will review shortly.

**Comment by ammar** [comment]
There's an extra space in the magic comment, after `:false;`, but it's still recognized by the ruby parser.

Looks good. Happy this is being fixed, finally.

When I looked at this a while back I used `ruby -i -pe`, but I prefer the less cryptic `open` and `write` approach:
```ruby
sh "ruby -i -pe 'BEGIN { gets; print \"# -*- warn-indent:false; -*-\n\" + $_ }; ' #{file}"
```

Planning on releasing a new version, 0.3.4, with a syntax file for ruby 2.3.1. Hopefully in the next couple of days.

Thanks & Cheers

File: Rakefile [hunk 1]
```diff
@@ -53,8 +53,17 @@ namespace :ragel do
   desc "Process the ragel source files and output ruby code"
   task :rb do |t|
     RAGEL_SOURCE_FILES.each do |file|
+      output_file = "#{RAGEL_OUTPUT_DIR}/#{file}.rb"
       # using faster flat table driven FSM, about 25% larger code, but about 30% faster
-      sh "ragel -F1 -R #{RAGEL_SOURCE_DIR}/#{file}.rl -o #{RAGEL_OUTPUT_DIR}/#{file}.rb"
+      sh "ragel -F1 -R #{RAGEL_SOURCE_DIR}/#{file}.rl -o #{output_file}"
```
Comment (dkubb) [inline Rakefile]: I realize this was merged, but I noticed this and `sh` can take an array and handle quoting properly, so you can write this as:

```ruby
sh %W[ragel -F1 -R #{RAGEL_SOURCE_DIR}/#{file}.rl -o #{output_file}]
```

Also, you could probably even simplify this further to:

```ruby
RAGEL_SOURCE_FILES.each do |file|
  output_file = Pathname.new("#{RAGEL_OUTPUT_DIR}/#{file}.rb")

  sh %W[ragel -F1 -R #{RAGEL_SOURCE_DIR}/#{file}.rl -o #{output_file}]

  contents = output_file.read
  output_file.write("# -*- warn-indent:false; -*-\n#{contents}")
end
```

**Comment by backus** [comment]
Thanks @ammar!

File: Rakefile (hunk 1 above)
Comment (ammar) [inline Rakefile]: Nice. I didn't know `sh` accepts an array. I like that.

**Comment by backus** [comment]
> Planning on releasing a new version, 0.3.4, with a syntax file for ruby 2.3.1. 

Any chance you could release 0.3.4 this week?

**Comment by ammar** [comment]
0.3.4 was released today.

**Pull request merged**